src/popover/container.tsx

```diff
@@ -47,145 +48,193 @@ export default function PopoverContainer({
   const popoverRef = useRef<HTMLDivElement | null>(null);
   const arrowRef = useRef<HTMLDivElement | null>(null);
+  const internalPositionRef = useRef<InternalPosition | null>(null);
 
   const [popoverStyle, setPopoverStyle] = useState<CSSProperties>(INITIAL_STYLES);
-  const [internalPosition, setInternalPosition] = useState<InternalPosition | null>(null);
   const isRefresh = useVisualRefresh();
 
   // Store the handler in a ref so that it can still be replaced from outside of the listener closure.
   const positionHandlerRef = useRef<() => void>(() => {});
 
+  // Remember when the popover was clicked in order to prevent eventually undesired position recalculations coming from that interaction.
+  const clickedInside = useRef(false);
+
   // Updates the position handler.
-  const updatePositionHandler = useCallback(() => {
-    if (!trackRef.current || !popoverRef.current || !bodyRef.current || !contentRef.current || !arrowRef.current) {
-      return;
-    }
-
-    // Get important elements
-    const popover = popoverRef.current;
-    const body = bodyRef.current;
-    const arrow = arrowRef.current;
-    const document = popover.ownerDocument;
-    const track = trackRef.current;
-
-    // If the popover body isn't being rendered for whatever reason (e.g. "display: none" or JSDOM),
-    // or track does not belong to the document - bail on calculating dimensions.
-    if (popover.offsetWidth === 0 || popover.offsetHeight === 0 || !nodeContains(document.body, track)) {
-      return;
-    }
-
-    // Imperatively move body off-screen to give it room to expand.
-    // Not doing this in React because this recalculation should happen
-    // in the span of a single frame without rerendering anything.
-    const prevTop = popover.style.top;
-    const prevLeft = popover.style.left;
-    popover.style.top = '0';
-    popover.style.left = '0';
-    // Imperatively remove body styles that can remain from the previous computation.
-    body.style.maxHeight = '';
-    body.style.overflowX = '';
-    body.style.overflowY = '';
-
-    // Get rects representing key elements
-    // Use getComputedStyle for arrowRect to avoid modifications made by transform
-    const viewportRect = getViewportRect(document.defaultView!);
-    const trackRect = track.getBoundingClientRect();
-    const arrowRect = {
-      width: parseFloat(getComputedStyle(arrow).width),
-      height: parseFloat(getComputedStyle(arrow).height),
-    };
-    const containingBlock = getContainingBlock(popover);
-    const containingBlockRect = containingBlock ? containingBlock.getBoundingClientRect() : viewportRect;
-
-    const bodyBorderWidth = getBorderWidth(body);
-    const contentRect = contentRef.current.getBoundingClientRect();
-    const contentBoundingBox = {
-      width: contentRect.width + 2 * bodyBorderWidth,
-      height: contentRect.height + 2 * bodyBorderWidth,
-    };
+  const updatePositionHandler = useCallback(
+    (onResize = false) => {
```

Comment:
Adding this `onResize` parameter made this entire function block appear in the diff because of the indentation. The actual relevant changes in it are lines 120 and 132.